proto/ibc/applications/middleware/fee/v1/query.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
+
+package ibc.applications.middleware.fee.v1;
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+import "ibc/applications/middleware/fee/v1/fee.proto";
+import "google/api/annotations.proto";
+import "ibc/core/channel/v1/channel.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+
+option go_package = "github.com/cosmos/ibc-go/modules/apps/29-fee/types";
+
+// Query provides defines the gRPC querier service.
+service Query {
+  // Gets the fee expected for submitting ReceivePacket msg for the given packet
+  rpc ReceiveFee(QueryReceiveFeeRequest) returns (QueryReceiveFeeResponse) {
+    option (google.api.http).get = "/ibc/apps/middleware/fee/v1/receive_fee";
+  }
+
+  // Gets the fee expected for submitting AcknowledgePacket msg for the given packet
+  rpc AckFee(QueryAckFeeRequest) returns (QueryAckFeeResponse) {
+    option (google.api.http).get = "/ibc/apps/middleware/fee/v1/ack_fee";
+  }
+
+  // Gets the fee expected for submitting TimeoutPacket msg for the given packet
+  rpc TimeoutFee(QueryTimeoutFeeRequest) returns (QueryTimeoutFeeResponse) {
+    option (google.api.http).get = "/ibc/apps/middleware/fee/v1/timeout_fee";
+  }
+
+  // Gets all incentivized packets
+  rpc IncentivizedPackets(QueryIncentivizedPacketsRequest) returns (QueryIncentivizedPacketsResponse) {
+    option (google.api.http).get = "/ibc/apps/middleware/fee/v1/incentivized_packets";
```

Comment:
These RPC URLs need to take in the arguments sent in the request. See the examples here: https://github.com/cosmos/ibc-go/blob/main/proto/ibc/core/channel/v1/query.proto#L18

As part of this, I'm realizing you may have to unravel the PacketId in the query request and place the strings at the top level of the request field so they can be properly parsed/unparsed from the URL. Sorry!

Reply:
perhaps there's a way to reference the channel id in the URL as `{packet_id.channel_id}`? But I've never tried so not sure if it will work

Reply:
I would look at how SDK handles this

Reply:
`{packet_id.channel_id}` etc worked.

Reply:
Updated :+1: